core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -148,14 +161,21 @@ private ZooKeeper getZooKeeper() {
   }
 
   private class ZCacheWatcher implements Watcher {
+
     @Override
     public void process(WatchedEvent event) {
       if (log.isTraceEnabled()) {
         log.trace("{}", event);
       }
+      setWatcherForTableConfigVersion(event);
 
       switch (event.getType()) {
         case NodeDataChanged:
+          Matcher tableConfigsMatcher = TABLE_CONFIGS_PATTERN.matcher(event.getPath());
+          if (tableConfigsMatcher.matches()) {
```

Comment:
We do not need to check for matches both outside and inside the processTableConfigurationItem method.  choose one or the other

Reply:
I will check it inside the function.  Thank you.  Good point.

Reply:
I pushed a new version.  I just got rid of the TABLE_CONFIGS_PATTERN and do 

`
(event.getPath().endsWith(Constants.ZTABLE_CONFIG_VERSION))
`
in the case statement.  Simpler and computationally less expensive.